chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/storemiddlewares/dataMaskingMiddleware.ts

```diff
@@ -37,7 +37,13 @@ const applyDataMasking = (action: IWebChatAction, regexCollection: IDataMaskingR
                 // eslint-disable-next-line no-cond-assign
                 while (match = regex.exec(text)) {
                     const replaceStr = match[0].replace(/./gi, maskedChar);
-                    text = text.replace(match[0], replaceStr);
+                    const modifiedText = text.replace(match[0], replaceStr);
```

Comment:
Same comment as V1 

Just wondering if we need to add anything on Telemetry side unless it is not flooding the system.

Like based on some boolean flag and and outside the loop

Reply:
For V2 is it hard to log it only once, since the input rules are not static and there's no good way to hold a global variable.

There's log if data masking rule DOES apply, so I think this will suffice